zkevm-circuits/src/copy_circuit.rs

```diff
@@ -667,14 +663,18 @@ impl<F: Field> CopyCircuitConfig<F> {
     }
 }
 
-/// Struct for test data, specifies values for related lookup tables
-#[cfg(test)]
+/// Struct for external data, specifies values for related lookup tables
 #[derive(Clone, Debug, Default)]
-pub struct CopyCircuitTestData {
+pub struct ExternalData {
+    /// TxCircuit -> max_txs
     pub max_txs: usize,
+    /// TxCircuit -> txs
     pub txs: Vec<Transaction>,
+    /// StateCircuit -> max_rws
     pub max_rws: usize,
```

Comment:
Isn't this already in the `CircuitParams` struct?? Same happens for `max_txs` no?? Is there a way to unify these things and reduce duplicity?

Reply:
`max_rws` and `max_txs` are indeed in `CircuitParams`.  Nevertheless after https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1102 the `witness::Block` (which contains `CircuitParams`) is no longer stored inside `CopyCircuit`!  That's why these fields are now here explicitly.  So there should be no code duplicity.
Also, the reason why `ExternalData` contains these two fields instead of a copy of `block.circuit_params` is to just store what's strictly necessary for running this circuit standalone.

Reply:
Ohhh my bad!!!! Thanks for the explanation!